addons/mail/static/src/new/web/messaging_menu/notification_item.js

```diff
@@ -26,4 +32,12 @@ export class NotificationItem extends Component {
     onClick(ev) {
         this.props.onClick(ev.target === this.markAsReadRef.el);
     }
+
+    get hasLeftSwipe() {
+        return this.props.onSwipeLeft && isMobileOS();
+    }
+
+    get hasRightSwipe() {
+        return this.props.onSwipeRight && isMobileOS() && this.props.hasMarkAsReadButton;
```

Comment:
can't we move up these conditions to before passing `onSwipeLeft` and `onSwipeRight` so that we don't need an extra getter and condition all around the code?